src/cljdoc/server/search/search.clj

```diff
@@ -185,6 +193,39 @@
               :doc (.doc (:searcher %) (.-doc h))})
            (:score-docs %))))))
 
+(defonce all-docs-cache (atom nil)) ;; As of 11/2019 it takes ± 2.6MB (well, when stringified)
+
+(defn all-docs
+  "Returns all the documents in the Lucene index, with all the versions.
+  NOTE: Takes a few seconds."
+  [^String index-dir]
```

Comment:
Could we extend this to already take group and artifact IDs? Then we would not have to filter the entire data in `available-docs-denormalized`

Reply:
It could but then we should create 3 fns - for all, for group, for artifact. And maybe give up on caching and just read the raw index every time, b/c it would be too complicated for all 3 cases?

What is the gain you seek? Performance? Or simpler code by making Lucene do the filtering? But the complexity just moves to here it doesn't really disappear I think.

I do return all documents because I believe this is what we do for the normal `/versions/*` requests as well. No? However I am open to anything...

Reply:
> What is the gain you seek? Performance? Or simpler code by making Lucene do the filtering? But the complexity just moves to here it doesn't really disappear I think.

I was wondering if maybe if we constrain at the Lucene level the more common queries (to get group/artifact versions) would be much faster than the current `all-docs` implementation.